packages/esbuild/src/executors/esbuild/lib/build-esbuild-options.ts

```diff
@@ -291,7 +291,16 @@ require('${mainFile}');
 }
 
 function getPrefixLength(pattern: string): number {
-  return pattern.substring(0, pattern.indexOf('*')).length;
+  const prefixIfWildcard = pattern.substring(0, pattern.indexOf('*')).length;
+  const prefixWithoutWildcard = pattern.substring(
+    0,
+    pattern.lastIndexOf('/')
+  ).length;
+  // if the pattern doesn't contain '*', then the length is always 0
+  // This causes issues when there are sub packages such as
+  // @nx/core
+  // @nx/core/testing
+  return prefixIfWildcard || prefixWithoutWildcard;
```

Comment:
I would just inline `prefixIfWildcard` and `prefixWithoutWildcard`, and it'll short-circuit the latter if previous is > 0.